src/clev2er/algorithms/seaice/alg_crop_waveform.py

```diff
@@ -0,0 +1,165 @@
+         .finalize() : Algorithm finalization/closure function (run after all chain
+                       processing completed)
+
+    Inherits from BaseAlgorithm which handles interaction with the chain controller run_chain.py
+
+    """
+
+    def init(self) -> Tuple[bool, str]:
+        """Algorithm initialization function
+
+        Add steps in this function that are run once at the beginning of the chain
+        (for example loading a DEM or Mask)
+
+        Returns:
+            (bool,str) : success or failure, error string
+
+        Test for KeyError or OSError exceptions and raise them if found
+        rather than just returning (False,"error description")
+
+        Raises:
+            KeyError : for keys not found in self.config
+            OSError : for any file related errors
+
+        Note:
+        - retrieve required config data from self.config dict
+        - log using self.log.info(), or self.log.error() or self.log.debug()
+
+        """
+        self.alg_name = __name__
+        self.log.info("Algorithm %s initializing", self.alg_name)
+
+        # --- Add your initialization steps below here ---
+
+        # --- End of initialization steps ---
+
+        return (True, "")
+
+    @Timer(name=__name__, text="", logger=None)
+    def process(self, l1b: Dataset, shared_dict: dict) -> Tuple[bool, str]:
+        """Main algorithm processing function, called for every L1b file
+
+        Args:
+            l1b (Dataset): input l1b file dataset (constant)
+            shared_dict (dict): shared_dict data passed between algorithms. Use this dict
+                                to pass algorithm results down the chain or read variables
+                                set by other algorithms.
+
+        Returns:
+            Tuple : (success (bool), failure_reason (str))
+            ie
+            (False,'error string'), or (True,'')
+
+        Note:
+        - retrieve required config data from self.config dict (read-only)
+        - retrieve data from other algorithms from shared_dict
+        - add results,variables from this algorithm to shared_dict
+        - log using self.log.info(), or self.log.error() or self.log.debug()
+
+        """
+
+        # This step is required to support multi-processing. Do not modify
+        success, error_str = self.process_setup(l1b)
+        if not success:
+            return (False, error_str)
+
+        # -------------------------------------------------------------------
+        # Perform the algorithm processing, store results that need to be passed
+        # \/    down the chain in the 'shared_dict' dict     \/
+        # -------------------------------------------------------------------
+
+        # bin_shift = original_cetre - ((bin_max - 50) + (128/2))
+        bin_shift = (shared_dict["waveform"].shape[-1] / 2) - (
+            (np.nanargmax(shared_dict["waveform"], axis=1) + 14)
+        )
```

Comment:
You should consider putting tunable constants in the algorithm configuration (for this algorithm). This makes it easier to reuse the chain for other missions which might have different waveform dimensions. So in the case they would go in config/chain_configs/seaice/seaice_A001.yml (or xml) as

alg_crop_waveform:
    some_setting: 128

Reply:
Changed this and added config for alg_flag_filters.py as well.